### PR TITLE
#3582

### PR DESCRIPTION
**Pull request by chengstory**

### CODE REVIEW

File: cocos/editor-support/cocostudio/CCComAudio.h [hunk 1]
```diff
@@ -56,6 +58,7 @@ class ComAudio : public cocos2d::Component
     virtual void onExit();
     virtual bool isEnabled() const;
     virtual void setEnabled(bool b);
+    virtual bool serialize(void* r);
```
Comment (dumganhar) [inline cocos/editor-support/cocostudio/CCComAudio.h]: please add `override` keyword for override functions.

File: cocos/editor-support/cocostudio/CCComRender.cpp [hunk 2]
```diff
@@ -67,6 +71,135 @@ void ComRender::setNode(cocos2d::Node *node)
...
+		{
+			if (strcmp(className, "CCSprite") == 0 && filePath.find(".png") != filePath.npos)
+			{
+				_render = Sprite::create(filePath.c_str());
+			}
+			else if(strcmp(className, "CCTMXTiledMap") == 0 && filePath.find(".tmx") != filePath.npos)
+			{
+				_render = TMXTiledMap::create(filePath.c_str());
+			}
+			else if(strcmp(className, "CCParticleSystemQuad") == 0 && filePath.find(".plist") != filePath.npos)
```
Comment (dumganhar) [inline cocos/editor-support/cocostudio/CCComRender.cpp]: `filePath.npos` --> `std::string::npos`

File: cocos/editor-support/cocostudio/CCComRender.cpp [hunk 3]
```diff
@@ -67,6 +71,135 @@ void ComRender::setNode(cocos2d::Node *node)
...
+			plistPath.assign(cocos2d::CCFileUtils::getInstance()->fullPathForFilename(plist));
+		}
+		int resType = DICTOOL->getIntValue_json(fileData, "resourceType", -1);
+		if (resType == 0)
+		{
+			if (strcmp(className, "CCSprite") == 0 && filePath.find(".png") != filePath.npos)
+			{
+				_render = Sprite::create(filePath.c_str());
+			}
+			else if(strcmp(className, "CCTMXTiledMap") == 0 && filePath.find(".tmx") != filePath.npos)
```
Comment (dumganhar) [inline cocos/editor-support/cocostudio/CCComRender.cpp]: `filePath.npos` --> `std::string::npos`

File: cocos/editor-support/cocostudio/ObjectFactory.h [hunk 4]
```diff
@@ -50,7 +49,8 @@ class ObjectFactory
 
     static ObjectFactory* getInstance();
     static void destroyInstance();
-    cocos2d::Object* createObject(const char *name);
+    cocos2d::Object* createObject(const std::string &name);
+    cocos2d::Component* createComponent(std::string name);
```
Comment (dumganhar) [inline cocos/editor-support/cocostudio/ObjectFactory.h]: Please use `const std::string&` instead.

File: cocos/editor-support/cocostudio/ObjectFactory.cpp [hunk 5]
```diff
@@ -100,6 +100,44 @@ Object* ObjectFactory::createObject(const char *name)
...
+    {
+        name = "ComController";
+    }
+    else if (name == "CCComAttribute")
+    {
+        name = "ComAttribute";
+    }
+    else if (name == "CCScene")
+    {
+        name = "CCScene";
```
Comment (dumganhar) [inline cocos/editor-support/cocostudio/ObjectFactory.cpp]: Should also remove `CC` prefix?

Comment (chengstory) [inline cocos/editor-support/cocostudio/ObjectFactory.cpp]: CCXXXX  is exported by CocoStudio,  support For  cocos2dx 2.x and cocos2dx 3.x.

**Pull request merged**